src/UploadImage.tsx

```diff
@@ -11,27 +11,32 @@ interface Props {
   ) => void;
 }
 
-export default class Upload3DImage extends Component<Props> {
+export default class UploadImage extends Component<Props> {
   private imageFileInfo: ImageFileInfo | null;
 
-  private slicesData: Array<Array<ImageBitmap>>;
+  private slicesData: Array<Array<ImageBitmap>> = [];
 
   constructor(props: Props) {
     super(props);
     this.imageFileInfo = null;
   }
 
   private uploadImage = (imageFile: File): void => {
-    this.readFile(imageFile)
-      .then((buffer: ArrayBuffer) => {
-        this.imageFileInfo = new ImageFileInfo(imageFile.name);
-        this.loadImageFile(buffer).catch((error) => {
+    const patternTiff = /tiff|tif$/i;
```

Comment:
I presume these are not case sensitive?

Reply:
Yeah, that's what the `i` on the end means

Reply:
Shows my shocking regex knowledge then!